doc: update README.md (#2786)
longhorn/longhorn-manager: README.md

Manager for Longhorn.

## Requirements

- Kubernetes cluster:
  - Kubernetes v1.21 or later is installed.
  - [Mount propagation](https://kubernetes-csi.github.io/docs/deploying.html#enabling-mount-propagation) is enabled.
- Host:
  - `iscsiadm`/`open-iscsi` and `nfs-common`/`nfs-utils`/`nfs-client` are installed.
  - The filesystem (ext4 or XFS) supports the `file extents` feature for data storage.

Run the [environment_check.sh](https://raw.githubusercontent.com/longhorn/longhorn/master/scripts/environment_check.sh) script to check if your system meets the listed requirements. For more information, see the [Longhorn documentation](https://longhorn.io/docs/latest/deploy/install/#installation-requirements).

## Build

`make`

## Deployment

- `kubectl create -f https://raw.githubusercontent.com/longhorn/longhorn/master/deploy/longhorn.yaml`

It will deploy the following components in the `longhorn-system` namespace:

> - Longhorn Manager

Author: Added. Thanks.

Manager for Longhorn.

## Requirements

> - Kubernetes cluster:

Reviewer: nit: The other way is to refer to the latest doc instead of duplicating the requirements here again.

> ## Requirement

Author: Modified. Thanks

> - Existing Kubernetes Cluster 1.21+.
> - [Mount propagation](https://kubernetes-csi.github.io/docs/deploying.html#enabling-mount-propagation) must be enabled.
> - Make sure `iscsiadm`/`open-iscsi` and `nfs-common`/`nfs-utils`/`nfs-client` has been installed on the host.
>   - Recommend that you execute the [environment_check.sh](https://raw.githubusercontent.com/longhorn/longhorn/master/scripts/environment_check.sh) script to check if all requirements have been fulfilled.
> - The host filesystem supports the `file extents` feature to store the data. Currently we support:
>   - ext4
>   - XFS
> 
> For more details of the installation requirements, please refer to the [Longhorn documentation.](https://longhorn.io/docs/latest/deploy/install/#installation-requirements)

Reviewer 2: ```suggestion
- Kubernetes cluster: 
  - Kubernetes v1.21 or later is installed.
  - [Mount propagation](https://kubernetes-csi.github.io/docs/deploying.html#enabling-mount-propagation) is enabled.
- Host:
  - `iscsiadm`/`open-iscsi` and `nfs-common`/`nfs-utils`/`nfs-client` are installed.
  - The filesystem (ext4 or XFS) supports the `file extents` feature for data storage.

Run the [environment_check.sh](https://raw.githubusercontent.com/longhorn/longhorn/master/scripts/environment_check.sh) script to check if your system meets the listed requirements. For more information, see the [Longhorn documentation](https://longhorn.io/docs/latest/deploy/install/#installation-requirements).
```

Author: Modified. Thanks, it's more clear.

## Build

`make`

## Deployment

- `kubectl create -f https://raw.githubusercontent.com/longhorn/longhorn/master/deploy/longhorn.yaml`

It will deploy the following components in the `longhorn-system` namespace:

> - Longhorn Managers

Reviewer 3: Add Longhorn CSI plugin components

Author: Added. Thanks

> - Longhorn Flexvolume Driver for Kubernetes

Reviewer 3: `Longhorn Flexvolume Driver for Kubernetes` is not supported, right?

Author: Removed. Thanks.

> - Longhorn Managers
> - Longhorn Flexvolume Driver for Kubernetes
> - Longhorn UI
> 
> For more details of the installation requirements, please refer to the [Longhorn documentation.](https://longhorn.io/docs/latest/deploy/install/)

Reviewer 2: ```suggestion
- Longhorn Manager
- Longhorn FlexVolume Driver for Kubernetes
- Longhorn UI

For more information, see the [Longhorn documentation](https://longhorn.io/docs/latest/deploy/install/).
```

Author: Modified. Thanks

## Cleanup

Longhorn CRD has finalizers in them, so user should delete the volumes and related resource first, give manager a chance to clean up after them.

To prevent damage to the Kubernetes cluster, we recommend deleting all Kubernetes workloads using Longhorn volumes (PersistentVolume, PersistentVolumeClaim, StorageClass, Deployment, StatefulSet, DaemonSet, etc).

> 1. Make sure the value of the Longhorn setting `deleting-confirmation-flag` is `true`.

Reviewer 2: ```suggestion
1. Ensure that the value of the Longhorn setting `deleting-confirmation-flag` is `true`.
```

Author: Modified. Thanks

```bash
kubectl -n longhorn-system edit setting deleting-confirmation-flag
```

> 2. Create the uninstallation job to cleanly purge CRDs from the system and wait for success:

Reviewer 2: ```suggestion
1. Create the uninstallation job to cleanly purge CRDs from the system. Verify that uninstallation was successful.
```

Author: Modified. Thanks

```bash
kubectl create -f https://raw.githubusercontent.com/longhorn/longhorn/master/uninstall/uninstall.yaml
kubectl get job/longhorn-uninstall -w
```

Example output:

```bash
$ kubectl create -f https://raw.githubusercontent.com/longhorn/longhorn/master/uninstall/uninstall.yaml
serviceaccount/longhorn-uninstall-service-account created
clusterrole.rbac.authorization.k8s.io/longhorn-uninstall-role created
clusterrolebinding.rbac.authorization.k8s.io/longhorn-uninstall-bind created
job.batch/longhorn-uninstall created

$ kubectl -n longhorn-system get job/longhorn-uninstall -w
NAME                 COMPLETIONS   DURATION   AGE
longhorn-uninstall   0/1           3s         3s
longhorn-uninstall   1/1           20s        20s
^C
```

> 3. Remove remaining components:

Reviewer 2: ```suggestion
1. Remove the remaining components.
```

Author: Modified. Thanks
